### ISSUE TITLE
Loiter engaged and quad starts gaining altitude.

### ISSUE THREAD

**Issue by RCJOSEB** [MegaPirateNG/ardupilot-mpng#10]
RCTimer V1.1 CRIUS AIOP
MegaPirateNG V3.0.1.R2

Hello Sir Alex:

I tested loiter with settings from last night that I had posted in rcgroups and again quad drifts and shoots up.
1. Erased eprom
2. Perform all calibrations
3. Changed Loiter PID from 1.0 to 0.5 per Arducopter warning
4. Left all other PID settings at their defaults
5. Performed a compassmot which resulted in less then 30% change

Tested loiter again. Same thing
1. Erased eprom
2. Perform all calibrations
3. Turned off auto declination and entered settings manually
4. Changed Loiter PID from 1.0 to 0.5 per Arducopter warning
5. Left all other PID settings at their defaults
6. Performed a compassmot which resulted in less then 30% change

Tested loiter again. Did a little better but still does this:
1. Engage loiter
2. Drops about a foot
3. Motors speed up
4. Rises a few feet and hovers a few seconds
5. Drops a little
6. Rises a few feet and keeps going and maybe stops to hover a few seconds

The baro is covered with foam and a small piece of black plastic is atop of the foam. All loiter tests are done at night.

In mission planner, the altitude starts with a negative number and fluctuates with the quad sitting still on the floor.  What is the best way to really test the baro.  I ran the test through the CLI and it read 0.0000m something on every line.

It never did anything like this in previous firmwares. Either I have a bad baro/crius or this firmware does not like me 

Any advice would be greatly appreciated.

Thanks,

Jose

**Comment by icsss** [comment]
I had a similar ALT HOLD behavior on 2.9.x
FIX: mounted the AIO controller on vibration dumpers. The inertial ALT HOLD is much more sensitive to vibrations.

**Comment by SirAlex** [comment]
As far as i know. It's vibration problem. There is lot information on DIYDrones topic about 3.0.1: http://diydrones.ning.com/forum/topics/arducopter-3-0-1-released

Since 3.0.1 uses inertial navigation for AltHold and position Hold modes. It's very important to decrease vibrations from motors/props. And also you must mount controller on dampening platform.

**Comment by RCJOSEB** [comment]
Thank you for the info icsss and Sir Alex!

**Comment by claytonc2001** [comment]
SirAlex, have you had a look at the gyro code to see if motor stopping bug is somehow connected yet? This bug needs squashing.

**Comment by SirAlex** [comment]
No, it's different. Gyro produces PANIC message in terminal once it hang, but our bug is another...

**Comment by thomasbutler** [comment]
Testing  v3.0.1r2 and had similar problem with loiter. I have ch6 set to position haold at 1500ms and loiter at 1800ms.  Quad position holds ok, but after a few seconds in loiter starts to increasingly gyrate.   Got it to the ground just before it would have flipped.

**Comment by thomasbutler** [comment]
Addditionally, flying in stabilize, quad control became sluggish and quad started to drift away with complete loss of control.  Crash damage too severe to do further testing today.

**Comment by thomasbutler** [comment]
Also, I got the quad going into gyrations on video!

**Comment by bigjake** [comment]
I am having what sound like similar issues.  Crius AIOP V2.  3.0.1 R2.
Crius board mounted on gel foam and suspended by O-rings on the corners. 
When switched to alt-hold, the motors surge to elevate copter about 2 feet then slow to drop the 2 feet and it repeats this over and over.  About a 2-3 second cycle. Is there a way to adjust PID's or other settings to change this behavior?  Can the GPS input for alt holding be modified?
Thanks

**Comment by dkillion** [comment]
GPS shouldn't be used for Alt-Hold, as the resolution for GPS is in the 2-3m (6-10 feet) range for accuracy.  

The Crius AIOP V2's barometric altimeter is accurate to 10cm (~4 inches), but only if it is properly protected from propwash and crosswinds with some open-cell foam placed over the sensor.

Does your altimeter have its foam?

**Comment by bigjake** [comment]
I do have the sensor covered with open cell foam.  Approx 1/4 inch thick and hot glued to the board on the left and right edges (off the sensor).  I asked about the GPS because others have stated it is used in alt holding.
Could this be a PID thing and some settings need to be changed?

Thanks.

**Comment by thomasbutler** [comment]
bigjake, my Crius was fine til  I went to v3xxx.  3x should have been a beta. we need to consider 3.xx a beta. No if ands or butts....

**Comment by thomasbutler** [comment]
Sorry Alex, but fact is 3xxx is not ready for prime time. If this had been software aircraft 99.999 no way flown! Will raise this issue with DIYDRONES

**Comment by SirAlex** [comment]
3.0.1-R3 already downgraded to Beta status.

But I'm make some fixes last time. I'm hope it will work fine now.

**Comment by thomasbutler** [comment]
SirAlex, sorry if my tone on that last comment was a bit rude.  All in all you've done a great job.  I should have been more carefull testing the latest version of MPNG.

**Comment by Homertje** [comment]
I've been flying my tricopter on 3.01 R3 for a few days now, and i had no problems like described... my Crius AIOP v1.1 is in the soft foam box i got when ordering the KK2.0 FC, added about 1/2" ESD proof SoftFoam on top of the baro.

**Comment by laflaf3d** [comment]
Hi,
thanks for your great job Sir Alex. I hear here some trouble that affect me too. Even with foam and light protection the althold mode is not accurate. And the RTL mode work but the landing is awefull! However, my real problem is the Poshold mode. I was on MPNG 3.1.5rc2 and my quadri don't accept PosHold mode. Channel 5 is at good value, GPS is 3D fix. But when I select PosHold, the quadri keep the last mode (stabilize or RTL...)... Why the life is so ruthless with me ? ;-p
